docs/dev/news/3958.fix

```diff
@@ -0,0 +1 @@
+Fix bad cast in Java target code
```

Comment:
I think you can drop this file, since this patch only affected unreleased code after 4.0 anyway?

Reply:
Oh, good point. I'll remove it.